docs/sources/operations/storage/tsdb.md

```diff
@@ -0,0 +1,77 @@
+---
+title: TSDB
+---
+# TSDB Index Store
+
+:warning: TSDB is still an experimental feature. It is not recommended to be used in production environments.
+
+With TSDB index store, you can run Grafana Loki without any dependency on NoSQL stores for storing index.
+It works the same way as [boltdb-shipper](./boltdb-shipper) index store, but with reduced TCO and better query performance.
+Loki uses [Prometheus TSDB format](https://github.com/prometheus/prometheus/tree/main/tsdb#tsdb) for storing the index with 
+some tweaks on top of it for query planning.
+
+## Example Configuration
+
+Example configuration with GCS:
+
+```yaml
+schema_config:
+  configs:
+    - from: 2018-04-15
+      store: tsdb
+      object_store: gcs
+      schema: v12
+      index:
+        prefix: loki_index_
+        period: 24h
+
+storage_config:
+  gcs:
+    bucket_name: GCS_BUCKET_NAME
+
+  tsdb_shipper:
+    active_index_directory: /loki/index
+    shared_store: gcs
+    cache_location: /loki/tsdb-cache
+```
+
+This would run Loki with TSDB index storing TSDB files locally at `/loki/index` and chunks at configured `GCS_BUCKET_NAME`.
+It would also keep shipping TSDB files periodically to the same configured bucket.
+It would also keep downloading TSDB files from shared bucket uploaded by other ingesters to `/loki/tsdb-cache` folder locally.
+
+## Operational Details
+
+Operationally, TSDB works the same way as BoltDB Shipper except the index format is different.
+All the [operational details for BoltDB Shipper](./boltdb-shipper#operational-details) apply to TSDB as well.
+However, there are some additional details to TSDB index store which are mentioned below.
+
+### Dynamic Query Sharding
+
+*Note:* Query sharding needs to be enabled using `parallelise_shardable_queries` under [query_range](../../configuration/_index#queryrange)
```

Comment:
Passive voice (enabled by who or what?).  Possible revisions:  "You must enable query sharding..."  or  "Your Loki Administrator must enable query sharding...."

Reply:
Would customers ever want to disable query sharding?  (I don't think they would, since it seems to be a useful feature.)  That's the only reason why I could see documenting how to enable/disable, since it's the default to be enabled.

Reply:
Query sharding is for speeding up queries by running more and smaller queries. I wanted to call out the flag here because I thought it was disabled by default and not all of the users would know about it and enable the feature. If someone is reading this block and has query sharding disabled, that means they would know how to enable it back.

Reply:
Were you going to remove this note, since query sharding is enabled by default?